Remove JAVA_HOME detection from the debian init script

nik9000 opened the pull request:
Only the debian init script did JAVA_HOME detection. Everything else just relied on `bin/elasticsearch`'s `which java` style detection. This strips the detection from the debian init script so its like the rpm init script.

Closes #13403
Closes #9774

nik9000 committed:
[packaging] remove JAVA_HOME detection

Only the debian init script did JAVA_HOME detection. Everything else just
relied on `bin/elasticsearch`'s `which java` style detection. This strips
the detection from the debian init script so its like the rpm init script.

Closes #13403

nik9000 commented:
Poke @tlrx and @spinscale for more to review. I don't know much of the history behind these detection lines and why we only have them in the debian init script and no where else but I suspect this is safe.

tlrx commented:
To be honest I think we should remove those lines and only checks for `java` bin availability and defaulting to JAVA_HOME if necessary

nik9000 commented:
> To be honest I think we should remove those lines and only checks for java bin availability and defaulting to JAVA_HOME if necessary

Like we do everywhere else? I can do that too. Its simple enough.

nik9000 commented:
> Like we do everywhere else? I can do that too. Its simple enough.

And done!

nik9000 commented:
@tlrx this one is ready for review as well.

tlrx commented:
@nik9000 Can you please add `closes #13514` to the commit message?

LGTM

nik9000 commented:
> @nik9000 Can you please add closes #9774 to the commit message?

I just squashed the commits and added the change to the commit message.

> LGTM

Merging then.
